sympy/galgebra/stringarrays.py

```diff
@@ -103,6 +103,5 @@ def str_combinations(base, lst, rank=1, mode='_'):
     forming the 'indexes' by concatenating combinations of elements from
     'lst' taken 'rank' at a time.
     """
-    str_lst = list(map(lambda x: base + mode + x, map(lambda x: reduce(operator.add, x),
-                        combinations(map(lambda x: str(x), lst), rank))))
+    str_lst = [base + mode + x for x in [reduce(operator.add, x) for x in combinations([str(x) for x in lst], rank)]]
```

Comment:
This is a prime example of doing too much in one line. You can leave it like this (this PR is just translating map to list comprehensions), or you can convert it to something a little easier to read. Personally, I would separate out each list comprehension into it's own variable, to make it more clear what's going on.